contracts/governance/Governor.sol

```diff
@@ -760,68 +761,24 @@ abstract contract Governor is Context, ERC165, EIP712, Nonces, IGovernor, IERC72
         address proposer,
         string memory description
     ) internal view virtual returns (bool) {
-        uint256 len = bytes(description).length;
+        uint256 length = bytes(description).length;
 
         // Length is too short to contain a valid proposer suffix
-        if (len < 52) {
+        if (length < 52) {
             return true;
         }
 
-        // Extract what would be the `#proposer=0x` marker beginning the suffix
-        bytes12 marker;
-        assembly ("memory-safe") {
-            // - Start of the string contents in memory = description + 32
-            // - First character of the marker = len - 52
-            //   - Length of "#proposer=0x0000000000000000000000000000000000000000" = 52
-            // - We read the memory word starting at the first character of the marker:
-            //   - (description + 32) + (len - 52) = description + (len - 20)
-            // - Note: Solidity will ignore anything past the first 12 bytes
-            marker := mload(add(description, sub(len, 20)))
-        }
+        // Extract what would be the `#proposer=` marker beginning the suffix
+        bytes10 marker = bytes10(_unsafeReadBytesOffset(bytes(description), length - 52));
```

Comment:
we can remove the private function and just have as it's only used once:
```suggestion
        bytes10 marker;
        assembly ("memory-safe") {
            marker := mload(add(description, sub(length, 52)))
        }
```

Reply:
length is at least 52 so the `sub` can't underflow

Reply:
lets mark the function unchecked then